### PR TITLE
Modernize Profile VM

### PR DESCRIPTION
**Pull request by luoser**
## what
Lots of input / output restructuring to clean up our ProfileViewModel, and oh hey add tests! Prep work for messages since this is an entry point.

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java [hunk 1]
```diff
@@ -33,20 +32,18 @@
...
   private RecyclerViewPaginator paginator;
 
-  protected @Bind(R.id.avatar) ImageView avatarImageView;
-  protected @Bind(R.id.user_name) TextView userNameTextView;
-  protected @Bind(R.id.created_num) TextView createdNumTextView;
-  protected @Bind(R.id.backed_num) TextView backedNumTextView;
-  protected @Bind(R.id.created) TextView createdTextView;
-  protected @Bind(R.id.backed) TextView backedTextView;
-  protected @Bind(R.id.divider) View dividerView;
+  protected @Bind(R.id.avatar_image_view) ImageView avatarImageView;
```
Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java]: nice nice, is this our style going forward? suffix to describe the UI component?

File: app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java [hunk 2]
```diff
@@ -62,25 +59,65 @@ protected void onCreate(final @Nullable Bundle savedInstanceState) {
...
-    viewModel.outputs.user()
+    viewModel.outputs.avatarImageViewUrl()
+      .compose(bindToLifecycle())
+      .observeOn(AndroidSchedulers.mainThread())
+      .subscribe(url -> Picasso.with(this).load(url).transform(new CircleTransformation()).into(avatarImageView));
+
+    viewModel.outputs.backedCountTextViewHidden()
+      .compose(bindToLifecycle())
+      .observeOn(AndroidSchedulers.mainThread())
+      .subscribe(ViewUtils.setGone(this.backedCountTextView));
```
Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java]: point freeeee allllll daaaaay

File: app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java [hunk 3]
```diff
@@ -16,85 +19,201 @@
...
+        Pair::create
+      )
+        .map(p -> p.first || p.second);
+
+      this.projects = paginator.paginatedData();
+      this.resumeDiscoveryActivity = this.exploreProjectsButtonClicked;
+      this.startProjectActivity = this.projectCardClicked;
+      this.userNameTextViewText = loggedInUser.map(User::name);
+
+      this.koala.trackProfileView();
```
Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java]: i suppose this is ok for now, but i think in general we only want to trigger this from a user event. in ios we often base this off of `viewDidLoad` or `viewDidAppear`. the equivalent in android might be when the `intent` is configured or something like that. something for us to discuss soon!

File: app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java [hunk 4]
```diff
@@ -16,85 +19,201 @@
...
+    private final Observable<String> backedCountTextViewText;
+    private final Observable<Boolean> backedTextViewHidden;
+    private final Observable<Boolean> createdCountTextViewHidden;
+    private final Observable<String> createdCountTextViewText;
+    private final Observable<Boolean> createdTextViewHidden;
+    private final Observable<Boolean> dividerViewHidden;
+    private final Observable<List<Project>> projects;
+    private final Observable<Void> resumeDiscoveryActivity;
+    private final Observable<Project> startProjectActivity;
+    private final Observable<String> userNameTextViewText;
```
Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java]: not a single behavior subject in sight!

File: app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java (hunk 1 above)
Comment (luoser) [inline app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java]: yep yep

File: app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java (hunk 4 above)
Comment (luoser) [inline app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java]: 😎

**Pull request merged**